### ISSUE TITLE
Add Geometry Offset

### ISSUE THREAD

**Issue by Leopold-Klotz** [xArm-Developer/xarm_ros2#77]
I'm having an issue with the "add_other_geometry" arguments for the package. I'm adding a rectangular mass (1.25kg) of length 20cm, width 10cm, height 10cm. This geometry can be successfully added; however, I am not finding a place in incorporate offset. The rectangle should not be centered on the end effector. In reality, my tool is oblong to one side and my TCP adjustments are: cx -45mm, cy 0mm, cz 43.5mm, x -140mm, y 0, z 43.5, roll 0, pitch -90deg, yaw 0. I ask this because there arguments for adjustments when added by mesh but I'm not seeing them for general geometry addition. 

In short, I'm wondering if there is a way to incorporate the offset of my tool with the "add_other_geometry" functionality, or if I should customize the urdf/srdf to incorporate this toll.

**Comment by gaspatxo** [comment]
Since you are using a builtin geomtetry, I cannot gurantee this works. I am using a custom stl file and for that the package offers the following possiblities:

`geometry_mesh_origin_xyz`
`geometry_mesh_origin_rpy`
`geometry_mesh_tcp_xyz`
`geometry_mesh_tcp_rpy`

Example: `geometry_mesh_tcp_rpy:='"0.0 0.0 1.5708"'`
